File: v1/example/demo_get_ntc_temperature.py
from fi_fsa import fi_fsa_v1
import time

server_ip_list = []


def main():
    server_ip_list = fi_fsa_v1.broadcast_func_with_filter(filter_type="Actuator")

    if server_ip_list:

        for i in range(len(server_ip_list)):
            temperature = fi_fsa_v1.get_ntc_temperature(server_ip_list[i])
            print(
                "mos_temperature = %f, armature_temperature = %f"
                % (temperature[0], temperature[1])
            )

        # end_time = time.time()
        #
        # # print the latency of the pvc function
        # latency = end_time - start_time
        # print(latency * 1000, "ms")

        time.sleep(0.2)  # sleep 1 second


if __name__ == "__main__":
    main()

File: v1/example/src/demo_get_ntc_temperature.cpp
#include "Fsa.h"
#include <iostream>

int main( int argc, const char** argv ) {
    string           ip = "192.168.137.101";
    FSA_CONNECT::FSA fsa;

    float mos_temperature      = 0;
    float armature_temperature = 0;

    fsa.init( ip );

    while ( true ) {
        fsa.GetNtcTemperature( mos_temperature, armature_temperature );

        std::cout << "mos_temperature: " << mos_temperature << "\t"
                  << "armature_tempature: " << armature_temperature << std::endl;

        sleep( 1 );
    }

    return 0;
}